edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -144,10 +151,11 @@ public async Task UpdatePriorities(IList<uint> priorities, Option<Endpoint> newE
             // Update priorities by merging the new ones with the existing.
             // We don't ever remove stale priorities, otherwise stored messages
             // pending for a removed priority will never get sent.
-            SortedDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
+            ImmutableDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
+            Dictionary<uint, EndpointExecutorFsm> updatedSnapshot = new Dictionary<uint, EndpointExecutorFsm>(snapshot);

```

Comment:
Why do we need this?

Reply:
we need to update dictionary and use immutable dictionary will create many copies.  Therefore I created a local copy of dictionary for update and convert to immutable dictionary once at the end.